pkg/cmd/tunnel/run.go

```diff
@@ -272,6 +272,52 @@ func (t *tunnelNodeReconciler) setupWithManager(
 		Complete(reconcile.Func(t.reconcile))
 }
 
+// buildTunnelClientOptions creates TunnelClientOptions from a TunnelNode.
+// Returns the options and server address, or an error with requeue result if the TunnelNode is not ready.
+func (t *tunnelNodeReconciler) buildTunnelClientOptions(ctx context.Context, tunnelNode *corev1alpha.TunnelNode) ([]tunnel.TunnelClientOption, string, *ctrl.Result, error) {
```

Comment:
instead of returning *ctrl.Result directly i'd return a custom error which wraps *ctrl.Result so that the refresh code which doesn't care about it can just look at the error.